README.md

```diff
@@ -2,17 +2,31 @@
 
 [![Build Status](https://travis-ci.org/kubernetes/test-infra.svg?branch=master)](https://travis-ci.org/kubernetes/test-infra)
 
-## Test Results History
+The test-infra repository contains a collection of tools for testing Kubernetes
+and displaying Kubernetes tests results. There is no merge bot here. Feel free
+to click the big green merge button once your code is reviewed and passes on
+Travis. You will need to be a member of
+[kubernetes/test-infra-maintainers](https://github.com/orgs/kubernetes/teams/test-infra-maintainers)
+to merge.
 
-The [Kubernetes test history
-dashboard](http://storage.googleapis.com/kubernetes-test-history/static/index.html)
-contains e2e test results from the last 24 hours.  The test history is
-implemented by two components in this repository:
+## Testing Kubernetes
+
+The YAML files under `jenkins/job-configs` define our Jenkins jobs via [Jenkins
+Job Builder](http://docs.openstack.org/infra/jenkins-job-builder/). Travis will
+run `jenkins/diff-job-config-patch.sh` to print out the XML diff between your
+change and master.
 
-1. `jenkins/test-history/` - scripts that periodically gather results from e2e
-   test jobs and generate the above status page.
-2. `gubernator/` - parses and presents the error message of a failed scenario.
-   The test-history page links failed scenarios to the output of gubernator.
+## Viewing Test Results
+
+* The [Kubernetes TestGrid](https://k8s-testgrid.appspot.com/) shows the results
+of test jobs for the last few weeks. It is currently not open-sourced, but we
+we would like to move in that direction eventually.
+* The [Kubernetes test history
+dashboard](http://storage.googleapis.com/kubernetes-test-history/static/index.html)
+collects test results from the last 24 hours. It is updated hourly by the
+scripts under `jenkins/test-history`.
+* [Gubernator](https://k8s-gubernator.appspot.com/) is an AppEngine app that
```

Comment:
`a Google App Engine site that parses and presents the results from individual test jobs.`